go/vt/vttablet/tabletmanager/vreplication/utils.go

```diff
@@ -0,0 +1,122 @@
+/*
+Copyright 2021 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package vreplication
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"vitess.io/vitess/go/sqltypes"
+	"vitess.io/vitess/go/vt/vtgate/evalengine"
+)
+
+const (
+	createVReplicationLog = `CREATE TABLE IF NOT EXISTS _vt.vreplication_log (
+		id BIGINT(20) AUTO_INCREMENT,
+		vrepl_id INT NOT NULL,
+		type VARBINARY(256) NOT NULL,
+		state VARBINARY(100) NOT NULL,
+		created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
+		updated_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
+		message text NOT NULL,
+		count BIGINT(20) NOT NULL DEFAULT 1,
+		PRIMARY KEY (id))`
+)
+
+const (
+	// Enum values for type column of _vt.vreplication_log
+
+	// LogStreamCreate is used when a row in _vt.vreplication is inserted via VReplicationExec
+	LogStreamCreate = "Stream Created"
```

Comment:
Is it possible/reasonable/idiomatic to make a proto enum for this? That way we also get a TypeScript enum generated for us. (Otherwise, I'll have to duplicate these strings on the front-end at some point.)

Reply:
I had punted on the enum: I think we discussed that this would be implemented as part of the vtadmin API around these logs.

Reply:
Revisiting at VTAdmin time makes sense to me!